src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -2051,84 +2055,107 @@ public String generateSummaryText(int numRecordsRead,
 		return summary;
 	}
 
+	public Collection<BulkLoadParentSaltFormLotDTO> getBulkLoadDTOsByBulkLoadFileId(Long bulkLoadFileId) {
+		// This query pulls back the corp name and bulk load file ids for all parents, saltforms, and lots that are at all linked
+		// to the bulkLoadFileId in question. 
+		EntityManager em = Parent.entityManager();
+        TypedQuery<BulkLoadParentSaltFormLotDTO> q = em.createQuery("SELECT new com.labsynch.labseer.dto.BulkLoadParentSaltFormLotDTO("
+		+ "p.corpName, p.bulkLoadFile.id, sf.corpName, sf.bulkLoadFile.id, l.corpName, l.bulkLoadFile.id) "
+		+ "FROM Lot l JOIN l.saltForm sf JOIN sf.parent p "
+		+ "WHERE l.bulkLoadFile.id = :bulkLoadFileId OR sf.bulkLoadFile.id = :bulkLoadFileId OR p.bulkLoadFile.id = :bulkLoadFileId",
+		BulkLoadParentSaltFormLotDTO.class);
+        q.setParameter("bulkLoadFileId", bulkLoadFileId);
+        return q.getResultList();
+	}
+
+	public Map<String, HashSet<String>> addToHashMapAsList(String key, String value, Map<String, HashSet<String>> map){
+		// Helper function to avoid repetition
+		if (map.containsKey(key)){
+			map.get(key).add(value);
+		} else {
+			HashSet<String> values = new HashSet<String>();
+			values.add(value);
+			map.put(key, values);
+		}
+		return map;
+	}
+
 	@Override
 	public PurgeFileDependencyCheckResponseDTO checkPurgeFileDependencies(BulkLoadFile bulkLoadFile) {
+		// acasDependencies is a map of corpName : list of ACAS experiments that depend on that corpName
 		Map<String, HashSet<String>> acasDependencies = new HashMap<String, HashSet<String>>();
+		// cmpdRegDependencies is a map of corpName : list of CReg BulkLoadFiles that depend on that corpName
 		Map<String, HashSet<String>> cmpdRegDependencies = new HashMap<String, HashSet<String>>();
+		// dependentSingleRegLots is a unique list of lot corpnames which came from single (non-bulk) registration and depend on parents or saltforms from this file
 		HashSet<String> dependentSingleRegLots = new HashSet<String>();
-		// we make a map with a CorpName as each key, and a list of dependent CorpNames
-		// as each value
-		int numberOfParents = 0;
-		int numberOfSaltForms = 0;
-		int numberOfLots = 0;
-		Collection<Parent> parents = Parent.findParentsByBulkLoadFileEquals(bulkLoadFile).getResultList();
```

Comment:
There are a lot of changes here, sorry! The main gist of this is:
Before: pull a list of Parents linked to this bulkLoadFile, then loop through its children and children's children, adding to the "acasDependencies", "cmpdRegDependencies", and "dependentSingleRegLots" Maps as we go.
Then repeat the same process with SaltForms (and their children), and then finally with Lots.

After: Do one big fetch of data using `getBulkLoadDTOsByBulkLoadFileId`, giving us all the info we need, then do some careful checking of the bulkLoadFile id columns for parent, saltform, and lot, and build up the same Map datastructures.